addons/hr_expense/models/hr_expense.py

```diff
@@ -917,6 +917,14 @@ def _compute_amount(self):
         for sheet in self:
             sheet.total_amount = sum(sheet.expense_line_ids.mapped('total_amount_company'))
 
+    @api.depends("account_move_id.line_ids")
+    def _compute_amount_residual(self):
+        for sheet in self:
+            residual_field = 'amount_residual'
+            payment_term_lines = sheet.account_move_id.sudo().line_ids \
+                .filtered(lambda line: line.expense_id in sheet.expense_line_ids and line.account_internal_type in ('receivable', 'payable'))
+            sheet.amount_residual = -sum(payment_term_lines.mapped(residual_field))
```

Comment:
This is clearer imo
```suggestion
            payment_term_lines = sheet.account_move_id.sudo().line_ids \
                .filtered(lambda line: line.display_type == 'payment_term')
            sheet.amount_residual = -sum(payment_term_lines.mapped('amount_residual'))
```

Reply:
`display_type` is not set in expense amls